client/server/user-bootstrap/test/index.js

```diff
@@ -0,0 +1,361 @@
+			cookies: {
+				wordpress_logged_in: 'auth-cookie',
+				support_session_id: 'session-id',
+			},
+			headers: {
+				'x-support-session': 'session-id',
+			},
+		} );
+
+		await expect( getBootstrappedUser( request ) ).rejects.toThrow(
+			new Error( 'Cannot bootstrap with both a support session header and support session cookie.' )
+		);
+	} );
+
+	it( 'sets the user agent in the upstream request', async () => {
+		const upstreamRequest = configureUpstreamRequest( {
+			headers: {
+				'User-Agent': 'WordPress.com Calypso',
+			},
+		} );
+		const request = mockRequest( {
+			cookies: {
+				wordpress_logged_in: 'auth-cookie',
+			},
+		} );
+
+		await getBootstrappedUser( request );
+
+		expect( upstreamRequest.isDone() ).toBe( true );
+	} );
+
+	it( 'sets the geo country code in the upstream request', async () => {
+		const upstreamRequest = configureUpstreamRequest( {
+			headers: {
+				'X-Forwarded-GeoIP-Country-Code': 'es',
+			},
+		} );
+		const request = mockRequest( {
+			cookies: {
+				wordpress_logged_in: 'auth-cookie',
+			},
+			headers: {
+				'x-geoip-country-code': 'es',
+			},
+		} );
+
+		await getBootstrappedUser( request );
+
+		expect( upstreamRequest.isDone() ).toBe( true );
+	} );
+
+	it( 'sets the auth session cookie', async () => {
+		const upstreamRequest = configureUpstreamRequest( {
+			headers: {
+				cookie: 'wordpress_logged_in=auth-cookie',
+			},
+		} );
+		const request = mockRequest( {
+			cookies: {
+				wordpress_logged_in: 'auth-cookie',
+			},
+		} );
+
+		await getBootstrappedUser( request );
+
+		expect( upstreamRequest.isDone() ).toBe( true );
+	} );
+
+	it( 'sets the support session cookie', async () => {
+		const upstreamRequest = configureUpstreamRequest( {
+			headers: {
+				cookie: ( val ) => val.includes( 'support_session_id=session-id' ),
+			},
+		} );
+		const request = mockRequest( {
+			cookies: {
+				wordpress_logged_in: 'auth-cookie',
+				support_session_id: 'session-id',
+			},
+		} );
+
+		await getBootstrappedUser( request );
+
+		expect( upstreamRequest.isDone() ).toBe( true );
+	} );
+
+	describe( 'with support session header', () => {
+		let request;
+		let upstreamRequest;
+
+		beforeEach( () => {
+			request = mockRequest( {
+				cookies: {
+					wordpress_logged_in: 'auth-cookie',
+				},
+				headers: {
+					'x-support-session': 'session-id',
+				},
+			} );
+			upstreamRequest = configureUpstreamRequest( {
+				headers: {
+					// Hardcoded value resulting from hashing "key" + "session-id"
+					Authorization: 'X-WPCALYPSO-SUPPORT-SESSION bc8844a622734cad18a2ec733e11b296',
+					'x-support-session': 'session-id',
+				},
+			} );
+		} );
+
+		it( 'throws if there is no support session API key', async () => {
+			withConfig( {
+				wpcom_calypso_support_session_rest_api_key: undefined,
+			} );
+
+			await expect( getBootstrappedUser( request ) ).rejects.toThrow(
+				new Error(
+					'Unable to boostrap user because of invalid SUPPORT SESSION API key in secrets.json'
```

Comment:
Would be nice to fix the "boostrap" typos here 🙂

Reply:
Fixed